### PR TITLE
serde sympy functions

### PR DESCRIPTION
**Pull request by avikchaudhuri**
Summary: Sympy functions appearing in symbolic expressions inside tensor metadata were not being deserialized properly.

Test Plan: updated test

Differential Revision: D60573150

### CODE REVIEW

File: torch/_export/serde/serialize.py [hunk 1]
```diff
@@ -1850,6 +1853,11 @@ def deserialize(
                 allow_non_fake_inputs=True,
                 shape_env=self.shape_env,
             )
+            self.sympy_functions = {
+                "ToFloat": torch.utils._sympy.functions.ToFloat,
+                "TruncToInt": torch.utils._sympy.functions.TruncToInt,
+                # add more sympy functions here
+            }
```
Comment (pianpwk) [inline torch/_export/serde/serialize.py:1860]: question, not comment: we need to add all functions here right? Not just the unsupported ones

Comment (avikchaudhuri) [reply torch/_export/serde/serialize.py:1860]: Yeah, but I don't know which ones should be here vs. handled elsewhere; some of them are int operations, like Mod and FloorDiv, should they be here? Or is it just the floor <-> int ones?

If you're going to do a pass anyway, maybe you can fill this up? Or let me know what else to add here.

Comment (avikchaudhuri) [reply torch/_export/serde/serialize.py:1883]: Added all.

**Review by pianpwk — APPROVED** [review]

**Pull request closed without merging**